src/chunk.c

```diff
@@ -3372,17 +3400,34 @@ ts_chunk_set_unordered(Chunk *chunk)
 	return ts_chunk_add_status(chunk, CHUNK_STATUS_COMPRESSED_UNORDERED);
 }
 
+/*No inserts,updates and deletes are permitted on a frozen chunk.
+ * Compression policies etc do not run on a frozen chunk.
+ * Only valid operation is dropping the chunk
+ */
 bool
-ts_chunk_add_status(Chunk *chunk, int32 status)
+ts_chunk_set_frozen(Chunk *chunk)
 {
-	return ts_chunk_set_status(chunk, ts_set_flags_32(chunk->fd.status, status));
+#if PG14_GE
+	return ts_chunk_add_status(chunk, CHUNK_STATUS_FROZEN);
+#else
+	return false;
+#endif
 }
 
-bool
-ts_chunk_set_status(Chunk *chunk, int32 status)
+static bool
+ts_chunk_add_status(Chunk *chunk, int32 status)
 {
-	chunk->fd.status = status;
-
+	if (ts_flags_are_set_32(chunk->fd.status, CHUNK_STATUS_FROZEN))
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+				 errmsg("chunk %d attempt to set status %d , current status %x ",
```

Comment:
Maybe more user-friendly error? Like simply `chunk is already freezed` or something like that, also we could add argument to the api function `if_not_freezed =>` to ignore it, if necessary